### PR TITLE
Enable multi-cidr deployment using config

### PR DESCRIPTION
**Pull request by suchita-g**
Signed-off-by: suchita-g <sgatfane@redhat.com>

### CODE REVIEW

**Comment by suchita-g** [comment]
PR Verification depends on the credential file defined in MR: https://gitlab.cee.redhat.com/ocs/ocs4-jenkins/-/merge_requests/1292

**Comment by suchita-g** [comment]
verification Job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-ms-provider-consumer/198

File: ocs_ci/deployment/rosa.py [hunk 1]
```diff
@@ -317,7 +317,7 @@ def host_network_update(self):
         security_group = self.aws.ec2_resource.SecurityGroup(sg_id)
         # The ports are not 100 % clear yet. Taken from doc:
         # https://docs.google.com/document/d/1RM8tmMbvnJcOZFdsqbCl9RvHXBv5K2ZI6ziQ-YTloGk/edit#
-        machine_cidr = config.ENV_DATA.get("machine-cidr", "10.0.0.0/16")
+        machine_cidr = config.ENV_DATA.get("machine_cidr", "10.0.0.0/16")
```
Comment (ebondare) [inline ocs_ci/deployment/rosa.py:320]: If you're changing `machine-cidr` to `machine_cidr`, then I suppose you'll need to change it also in conf/deployment/rosa/private_link_deployment.yaml

**Review by ebondare — CHANGES_REQUESTED** [review]

File: ocs_ci/deployment/rosa.py (hunk 1 above)
Comment (suchita-g) [reply ocs_ci/deployment/rosa.py:320]: no need of defining default cidr in the private link conf as it is considered in a code. updated private_link_deployment.yaml conf

File: conf/deployment/rosa/private_link_deployment.yaml [hunk 2]
```diff
@@ -1,4 +1,3 @@
 ---
 ENV_DATA:
   private_link: True
-  machine-cidr: '10.1.0.0/16'
```
Comment (ebondare) [inline conf/deployment/rosa/private_link_deployment.yaml:4]: This change needs a successful privatelink deployment job for verification. As far as I can see, the privatelink job was https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/20431/ but it failed

Comment (suchita-g) [reply conf/deployment/rosa/private_link_deployment.yaml:4]: This failed after deploying the provider cluster while accessing cluster. As I missed mentioning proxy
which result in ```2023-02-08 14:14:44  08:44:44 - MainThread - ocs_ci.utility.utils - WARNING  - Command stderr: Unable to connect to the server: dial tcp: lookup api.sgatfane-pl-pr.zn1a.s1.devshift.org on 10.11.5.160:53: no such host
2023-02-08 14:14:44  08:44:44 - MainThread - ocs_ci.ocs.openshift_ops - ERROR  - Cluster is not ready to use: Error during execution of the command: oc cluster-info.```
However, the cluster got deployed successfully,

This is today's job: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/20494/console

**Review by jilju — APPROVED** [review]

**Review by fbalak — APPROVED** [review]

**Review by ebondare — APPROVED** [review]
LGTM

**Pull request merged**